bundles/org.openhab.binding.robonect/src/main/java/org/openhab/binding/robonect/internal/handler/RobonectHandler.java

```diff
@@ -362,26 +362,19 @@ public void initialize() {
                     timeZoneString, timeZoneProvider.getTimeZone(), e);
         }
 
-        try {
-            httpClient.start();
-            robonectClient = new RobonectClient(httpClient, endpoint);
-        } catch (Exception e) {
-            logger.error("Exception while trying to start http client", e);
-            throw new RuntimeException("Exception while trying to start http client", e);
-        }
+        robonectClient = new RobonectClient(httpClient, endpoint);
         Runnable runnable = new MowerChannelPoller(TimeUnit.SECONDS.toMillis(robonectConfig.getOfflineTimeout()));
         int pollInterval = robonectConfig.getPollInterval();
         pollingJob = scheduler.scheduleWithFixedDelay(runnable, 0, pollInterval, TimeUnit.SECONDS);
     }
 
     @Override
     public void dispose() {
+        ScheduledFuture<?> pollingJob = this.pollingJob;
         if (pollingJob != null) {
             pollingJob.cancel(true);
-            pollingJob = null;
+            this.pollingJob = null;
         }
-
-        httpClient = null;
```

Comment:
IMHO the only thing you need to change in order to fix the configuration update of the `Thing` is to remove this line.

On a configuration update the `BaseThingHandler` calls `dispose` which sets the `httpClient` to `null` but afterwards it calls `initialize` which has no chance of retrieving the `httpClient` from the factory.

The other option would be to inject the `HttpClientFactory httpClientFactory` into the `ThingHandler` and call `httpClient = httpClientFactory.getCommonHttpClient()` in `initialize` and set it to `null` in `dispose` which would probable be the best solution.

I don't understand your comment regarding the collision with the `AuthenticationStore` which is why you decided to create a dedicated client. IMHO we have a shared one to be used with as much bindings as possible.

Reply:
> IMHO the only thing you need to change in order to fix the configuration update of the `Thing` is to remove this line.

For the minimal fix I agree, except this should also be removed:
https://github.com/openhab/openhab-addons/blob/93b888ad31bb97caa30adb0bd0218050d5773959/bundles/org.openhab.binding.robonect/src/main/java/org/openhab/binding/robonect/internal/handler/RobonectHandler.java#L366

since the binding should not manage the life-cycle of the shared HTTP client.

> The other option would be to inject the `HttpClientFactory httpClientFactory` into the `ThingHandler` and call `httpClient = httpClientFactory.getCommonHttpClient()` in `initialize` and set it to `null` in `dispose` which would probable be the best solution.

There is no reason to get the shared client in `initialize`, since it will be the same instance each time. So it can be set by the constructor when the handler is created.

> I don't understand your comment regarding the collision with the `AuthenticationStore` which is why you decided to create a dedicated client. IMHO we have a shared one to be used with as much bindings as possible.

I have to admit I'm not really sure about this part. As you can see in the quoted code the `AuthenticationStore` instance from the shared client is obtained, and an `AuthenticationResult` is added. I don't really know if this can interfere with other bindings, so wanted to be on the safe side. If this is not needed, I'll happily revert to using the shared client.

Reply:
> For the minimal fix I agree, except this should also be removed:
[openhab-addons/bundles/org.openhab.binding.robonect/src/main/java/org/openhab/binding/robonect/internal/handler/RobonectHandler.java](https://github.com/openhab/openhab-addons/blob/93b888ad31bb97caa30adb0bd0218050d5773959/bundles/org.openhab.binding.robonect/src/main/java/org/openhab/binding/robonect/internal/handler/RobonectHandler.java#L366)

I agree, because indeed the common client will be started here
 
https://github.com/openhab/openhab-core/blob/main/bundles/org.openhab.core.io.net/src/main/java/org/openhab/core/io/net/http/internal/WebClientFactoryImpl.java#L282

I overlooked that when creating the fix https://github.com/openhab/openhab-addons/pull/9429 back then.

Oh, right. You are talking about
https://github.com/openhab/openhab-addons/blob/main/bundles/org.openhab.binding.robonect/src/main/java/org/openhab/binding/robonect/internal/RobonectClient.java#L172-L174

Attaching some robonect specific credentials to the `common` client doesn't look right to me either.

So, yes please create a robonect specific `httpClient`, but PER `ThingHandler` and start/stop it there.
Because you can have 2 `RobonectHandlers` with different credentials each!

Reply:
Done - are you able to test with this change? The JAR linked in the PR description has been updated.

Reply:
Thanks, its a small change and looks good from the code, so it should be fine without extensive testing.

Reply:
Extensive testing is probably not needed, but at least I would like someone to test that the thing comes online and is able to communicate, and also that reinitialization works correctly, i.e. configuration changes. I have not tested anything, since I don't have a Robonect.

Reply:
I have just tested your branch in a local IDE against my Robonect and stepped through the configuration update code. There is no more NPE and the Thing is ONLINE and the items receive their updates.

So, I would say that it also looks good from a test perspective!